### PR TITLE
V4 Missing Data Message on Charts

### PR DESCRIPTION
**Pull request by devinmatte**
## Motivation

#471 

## Changes

Screenshot 2023-04-30 at 12 15 07 PM

### CODE REVIEW

File: common/components/charts/AggregateLineChart.tsx [hunk 1]
```diff
@@ -178,6 +179,9 @@ export const AggregateLineChart: React.FC<AggregateLineProps> = ({
             {
               id: 'customTitleAggregate',
               afterDraw: (chart: ChartJS) => {
+                if (startDate === undefined || endDate === undefined || data.length === 0) {
+                  writeError(chart);
```
Comment (PatrickCleary) [inline common/components/charts/AggregateLineChart.tsx:183]: [This upcoming PR](https://github.com/transitmatters/t-performance-dash/pull/514) should prevent this component from being loaded without a startDate or endDate, so I think we only need to check for data being an empty array

File: common/components/charts/SingleDayLineChart.tsx [hunk 2]
```diff
@@ -216,25 +215,13 @@ export const SingleDayLineChart: React.FC<SingleDayLineProps> = ({
...
-                  const { height } = chart;
-                  chart.clear();
-
-                  ctx.save();
-                  ctx.textAlign = 'center';
-                  ctx.textBaseline = 'middle';
-                  ctx.font = "16px normal 'Helvetica Nueue'";
-                  ctx.fillText('No data to display', width / 2, height / 2);
-                  ctx.restore();
+                if (date === undefined || date.length === 0 || data.length === 0) {
```
Comment (PatrickCleary) [inline common/components/charts/SingleDayLineChart.tsx:223]: Was there a case when we were passing an empty string for date? Or is this just out of precaution?

**Review by PatrickCleary — APPROVED** [review]
LGTM

File: common/components/charts/SingleDayLineChart.tsx (hunk 2 above)
Comment (devinmatte) [reply common/components/charts/SingleDayLineChart.tsx:223]: There was in v3, so this was handling those cases from there. It doesn't hurt to check

**Pull request merged**